### ISSUE TITLE
Deprecate `date_time_string` and `string_date_time` in twisted.web.server

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5535]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)| @washort reported|
|-|-|
|Trac ID|trac#5535|
|Type|task|
|Created|2012-03-12 23:14:03Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-webserver-attributes-5535|

In 2002, these were marked as "backwards compatibility". It's time to let go.

Attachments:

* [twisted5535.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e1a/e1a4edba2a1724899037c9366eb8389db3205b90/34d439996ec754988076eb9d895840eff1a6d17a.patch) (831 bytes) - added by necaris on 2012-03-12 23:21:56Z - Deprecate the mentioned attributes
* [twisted5535.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e1a/e1a4edba2a1724899037c9366eb8389db3205b90/b99aa16d7d5e9e6a2e68d206eef399ca8541205b.patch) (2501 bytes) - added by necaris on 2012-03-13 18:04:28Z - Updated patch with tests
* [twisted5535.3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e1a/e1a4edba2a1724899037c9366eb8389db3205b90/e7121592d40c92b9d3224d3077a52d3c9df6e4c6.patch) (3359 bytes) - added by necaris on 2012-03-13 20:52:25Z - Updated patch with docstrings and news file

<details><summary>Searchable metadata</summary>

```
trac-id__5535 5535
type__task task
reporter__washort washort
priority__normal normal
milestone__ 
branch__branches_deprecate_webserver_attributes_5535 branches/deprecate-webserver-attributes-5535
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1331594043000000 1331594043000000
changetime__1331675146000000 1331675146000000
version__None None
owner__ 
cc__jknight cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli set owner to necaris|
|-|-|

Thanks for working on this necaris!

Could you please add some unit tests for the deprecations?  I couldn't find the wiki page on how to test deprecations, but [here is an example](http://twistedmatrix.com/trac/attachment/ticket/5517/5517.diff) (although in that ticket, those tests were being removed).

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

Ok, found one wiki page on the [Compatibility Policy](http://twistedmatrix.com/trac/wiki/CompatibilityPolicy).

And rather than the deprecation message of "No longer used", can you change them to say what they are deprecated in favor of instead?  In this case, looks like they are just aliases for the http functions.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

And the version should be 12.1 instead of 12.2.

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|necaris removed owner|
|-|-|

Updated patch with tests, a better deprecation message, and corrected version

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli set owner to necaris|
|-|-|

Thanks for all your wonderful work, necaris!

This looks great, except that the tests (both the test_XX methods, as well as the class) need docstrings.

Also, is just a nitpick, but perhaps ServerAttributesTest should be named ServerAttributesTests, since it's more than 1 test, or ServerAttributesTestCase?

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

Oops, also, can you please also add a [news file](http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles)?

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|necaris removed owner|
|-|-|

Better docstrings and news file included in the latest patch -- thank you for your patience with me!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli commented|
|-|-|

(In [33746]) Branching to 'deprecate-webserver-attributes-5535'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_11"></a>|@cyli commented|
|-|-|

Meh, I forgot the refs tag in that last commit:  [33748]

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_12"></a>|@cyli commented|
|-|-|

Thanks for all your hard work and for being so responsive necaris!

This looks great ([buildbot results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-webserver-attributes-5535)) - I will merge this into trunk.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_13"></a>|@cyli set status to `closed`|
|-|-|

(In [33749]) Merge deprecate-webserver-attributes-5535: deprecate 'date_time_string' and 'string_date_time' in t.w.server

Author: necaris
Reviewer: cyli
Fixes: [#5535](https://github.com/twisted/twisted/issues/5535)

Deprecate `date_time_string` and `string_date_time` in twisted.web.server, which were marked as 'backwards compatibility' in 2002.

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_14"></a>|necaris commented|
|-|-|

cyli: Thank you for the patient reviews!
